### ISSUE TITLE
[browser][Codespaces] Build does not pass on Codespaces

### ISSUE THREAD

**Issue by ilonatommy** [dotnet/runtime#82942]
Building runtime on Codespaces, e.g. `./dotnet.sh build -os browser -c Debug` fails with:
![image](https://user-images.githubusercontent.com/32700855/222734081-13acbbdd-825d-4b63-a28f-37579ed38d51.png)

The manual fix is to go to set `UseHardlinksIfPossible=false` in the affected targets (`intelisende.targets`, `optimizationData.targets`).

**Comment by radical** [comment]
How do I reproduce this?

**Comment by lewing** [comment]
Looks like this is possibly a regression due to https://github.com/dotnet/msbuild/pull/8238 and an exception looking up the resource?

cc @rainersigwald @JaynieBai

**Comment by radical** [comment]
Trying to create the hardlink manually fails because it would be a cross device link.

**Comment by rainersigwald** [comment]
> Looks like this is possibly a regression due to [dotnet/msbuild#8238](https://github.com/dotnet/msbuild/pull/8238) and an exception looking up the resource?

Yes. And this combined with the cross-device link means we turn what should have been an info message "we tried to link but it failed so we fell back to actual copy" into an error.

I'll move to MSBuild.
